packages/organizations/origin-organization-irec-api/src/connection/handlers/create-connection.handler.ts

```diff
@@ -101,14 +101,16 @@ export class CreateConnectionHandler implements ICommandHandler<CreateConnection
                 AccountType.Issue,
                 AccountType.Trade
             ]);
-        } else if (isRegistrant || isParticipant) {
+        } else if (isParticipant) {
             await this.createAccounts(clientId, clientSecret, tokens, organization, [
                 AccountType.Trade
             ]);
         } else {
-            throw new BadRequestException(
-                'IREC account organization has to have issuer or registrant or participant role'
-            );
+            if (!isRegistrant) {
+                throw new BadRequestException(
+                    'IREC account organization has to have issuer or registrant or participant role'
+                );
+            }
```

Comment:
so being registrant doesn't create the accounts?

Reply:
Exactly, old implementation was not creating that as well, I refactored it in wrong way. Now it is correct

Reply:
Registrant cannot have trade account created